keg/web.py

```diff
@@ -237,8 +237,10 @@ def method_with_rules(obj):
                                             method_route.sanitized_method_name('_'))
                     cls.view_funcs[method_route.endpoint] = view_func
                 mr_options['view_func'] = cls.view_funcs[method_route.endpoint]
-                cls.blueprint.add_url_rule(method_route.rule(), **mr_options)
-            if method_name in http_method_funcs:
+
+                if cls.blueprint:
+                    cls.blueprint.add_url_rule(method_route.rule(), **mr_options)
+            if method_name in http_method_funcs and method_name.upper() in cls.methods:
```

Comment:
What was the purpose of adding `and method_name.upper() in cls.methods` to the if statement?